addon/mixins/component.js

```diff
@@ -0,0 +1,95 @@
+import Ember from 'ember';
+import layout from '../templates/sl-behavior';
+import resolveValue from '../utils/resolve-value';
+
+/**
+ * Adds ability to hide template content of a component under a given set of conditions
+ *
+ * @module
+ * @augments ember/Mixin
+ */
+export default Ember.Mixin.create({
+
+    // -------------------------------------------------------------------------
+    // Dependencies
+
+    // -------------------------------------------------------------------------
+    // Attributes
+
+    /** @type {Object} */
+    layout,
+
+    /** @type {String} */
+    tagName: 'span',
+
+    // -------------------------------------------------------------------------
+    // Actions
+
+    // -------------------------------------------------------------------------
+    // Events
+
+    // -------------------------------------------------------------------------
+    // Properties
+
+    /**
+     * The behavior service used to check if a supplied behavior is allowed
+     *
+     * @type {ember/Service}
+     */
+    behaviorService: Ember.inject.service( 'sl-behavior' ),
+
+    /**
+     * Indicates whether an activity is allowed if the behaviors set on the service allow it
+     *
+     * When set to a function, the return value must be a Boolean and will not
+     * be observed by the showContent() method.
+     *
+     * @type {Boolean|Function}
+     */
+    possible: true,
+
+    // -------------------------------------------------------------------------
+    // Observers
+
+    // -------------------------------------------------------------------------
+    // Methods
+
+    /**
+     * Implement this function to customize the block content's viewability
+     *
+     * @function
+     * @abstract
+     * @returns {Boolean} true if the content is viewable, otherwise false
+     */
+    isViewable() {
```

Comment:
I did have that originally but jsHint threw a fit since the params weren't being used. How would you suggest getting around that? I originally had this:
````
    isViewable( activity, resource, possible ) {
        Ember.assert( 'The mixins/component.isViewable() method should be implemented on the derived class' );

        return activity && resource && possible && false;
    },
````